node/node_newblock.go

```diff
@@ -84,6 +84,11 @@ func (node *Node) WaitForConsensusReadyV2(readySignal chan consensus.ProposalTyp
 					newBlock, err := node.ProposeNewBlock(newCommitSigsChan)
```

Comment:
also, since we don't want to propose the same block number, does it make sense to do the checking in ProposeNewBlock function?

Reply:
proposeNewBlock is more like a utility, it's for the caller to determine how to use it. So adding the check in the caller is fine. There is not much performance concern here. And it will be cumbersome to add check inside proposeNewBlock too due to its lack of easy access to the proposed block number.